src/api/starMessage.js

```diff
@@ -0,0 +1,7 @@
+/* @flow */
```

Comment:
Not a big fan of the naming conflict with the event handlers.
Why not call the API call 'toggleMessageStarred' and set the third parameter being boolean - true = star, false = unstar?